src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -0,0 +1,296 @@
+package com.minecolonies.coremod.colony.buildings.views;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.ReflectionUtils;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.colony.CitizenDataView;
+import com.minecolonies.coremod.colony.ColonyView;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_CITIZEN_BY_REQUEST;
+import static com.minecolonies.coremod.colony.buildings.AbstractBuilding.NO_WORK_ORDER;
+import static com.minecolonies.coremod.colony.buildings.AbstractBuilding.processIntegerKeyTokenList;
+
+/**
+ * The AbstractBuilding View is the client-side representation of a AbstractBuilding.
+ * Views contain the AbstractBuilding's data that is relevant to a Client, in a more client-friendly form.
+ * Mutable operations on a View result in a message to the server to perform the operation.
+ */
+public class AbstractBuildingView implements IRequester
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this class abstract or rename it, since it matches the regular expression '^Abstract[A-Z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00118)